src/js/calculator.js

```diff
@@ -0,0 +1,62 @@
+import { MAX_LENGTH } from "./const.js";
+
+class Calculator {
+	result = 0;
+	current = 0;
+	operator = "";
+
+	calculate = {
+		"+": () => this.add(),
+		"-": () => this.substract(),
+		X: () => this.multiply(),
+		"/": () => this.divide(),
+	};
+
+	add() {
+		this.result += Number(this.current);
+	}
+
+	substract() {
+		this.result -= Number(this.current);
+	}
+
+	multiply() {
+		this.result *= Number(this.current);
+	}
+
+	divide() {
+		this.result = Math.floor(this.result / Number(this.current));
+	}
+
+	allClear() {
+		this.current = 0;
+		this.result = 0;
+		this.operator = "";
+	}
+
+	pressDigit(digit) {
+		if (isOverMaxLength(this.current)) return;
+		this.current = Number(`${this.current}${digit}`);
+	}
+
+	pressOperator(currentOperator) {
+		if (currentOperator === "=") {
+			this.calculate[this.operator]();
+			this.current = this.result;
+		} else {
+			this.result = Number(this.current);
+			this.current = 0;
+		}
+
+		this.operator = currentOperator;
+	}
+}
+
+const isOverMaxLength = (number) => {
+	if (String(number).length >= MAX_LENGTH) {
+		alert(`${MAX_LENGTH}자리 이하의 수만 입력 가능합니다.`);
+		return true;
+	}
+};
```

Comment:
```suggestion
const isOverMaxLength = (number) => String(number).length >= MAX_LENGTH;
```
함수가 2가지 일을 하고있어요!
네이밍에 알맞는 동작만 남겨주는게 좋아요.
경고창을 띄워주는 역할은 호출하는쪽에 위임해줍시다~

Reply:
넵! 수정했습니다